packages/compiler/src/core/program.ts

```diff
@@ -830,6 +697,7 @@ export async function compile(
    * @param emitter Emitter ref to run
    */
   async function runEmitter(emitter: EmitterRef) {
+    const { createAssetEmitter } = await import("../emitter-framework/asset-emitter.js");
```

Comment:
This one is a bit anyoing but the alternative is that the asset emitter never loads things from `core/index.js` but always in small pieces

Reply:
I think we should avoid dynamic import if at all possible. Importing from leaves rather than barrel seems like a better approach (as annoying as that is).